rust/operator-binary/src/restart_controller.rs

```diff
@@ -0,0 +1,308 @@
+                    format!(
+                        "{}/{}",
+                        cm.metadata.uid.as_ref()?,
+                        cm.metadata.resource_version.as_ref()?
+                    ),
+                ))
+            }),
+    );
+    let secret_refs = pod_specs
+        .flat_map(|pod_spec| {
+            find_pod_refs(
+                pod_spec,
+                |volume| {
+                    Some(ObjectRef::<Secret>::new(
+                        volume.secret.as_ref()?.secret_name.as_deref()?,
+                    ))
+                },
+                |env_var| {
+                    Some(ObjectRef::<Secret>::new(
+                        env_var
+                            .value_from
+                            .as_ref()?
+                            .secret_key_ref
+                            .as_ref()?
+                            .name
+                            .as_deref()?,
+                    ))
+                },
+                |env_from| {
+                    Some(ObjectRef::<Secret>::new(
+                        env_from.secret_ref.as_ref()?.name.as_deref()?,
+                    ))
+                },
+            )
+        })
+        .map(|secret_ref| secret_ref.within(ns));
+    annotations.extend(
+        secret_refs
+            .flat_map(|secret_ref| ctx.get_ref().secrets.get(&secret_ref))
+            .flat_map(|cm| {
+                Some((
+                    format!(
+                        "secret.restarter.stackable.tech/{}",
+                        cm.metadata.name.as_ref()?
+                    ),
+                    format!(
+                        "{}/{}",
+                        cm.metadata.uid.as_ref()?,
+                        cm.metadata.resource_version.as_ref()?
+                    ),
+                ))
+            }),
+    );
+    let stses = kube::Api::<StatefulSet>::namespaced(ctx.get_ref().kube.clone(), ns);
+    stses
+        .patch(
+            &sts.name(),
+            &PatchParams {
+                force: true,
+                field_manager: Some("restarter.stackable.tech/statefulset".to_string()),
+                ..PatchParams::default()
+            },
+            &Patch::Apply(
+                // Can't use typed API, see https://github.com/Arnavion/k8s-openapi/issues/112
+                json!({
+                    "apiVersion": "apps/v1",
+                    "kind": "StatefulSet",
+                    "metadata": {
+                        "name": sts.metadata.name,
+                        "namespace": sts.metadata.namespace,
+                        "uid": sts.metadata.uid,
+                    },
+                    "spec": {
+                        "template": {
+                            "metadata": {
+                                "annotations": annotations,
+                            },
+                        },
+                    },
+                }),
+            ),
+        )
+        .await
+        .unwrap();
```

Comment:
This may fail? Can we have proper error handling here?

Reply:
I'm struggling to create the ApplySnafu error :/ Can you please give me a hint how to do that?
I have tried lot's of variants, this is my current approach
```
        .await
        .context(ApplySnafu {
            obj_ref: ObjectRef::from_obj(&sts)
        })?;
```